data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsDataFactory.java

```diff
@@ -33,38 +35,87 @@ public MetricsDataFactory(MetricsSystem metricsSystem) {
     this.metricsSystem = metricsSystem;
   }
 
-  public BaseMetricData getMetricData(MetricsDataClient metricsClient) {
+  public List<BaseMetricData> getMetricData() {
+    List<BaseMetricData> metricList = new ArrayList<>();
     if (metricsSystem instanceof PrometheusMetricsSystem) {
-      if (metricsClient == MetricsDataClient.VALIDATOR) {
-        return extractValidatorData((PrometheusMetricsSystem) metricsSystem);
-      }
+      Map<String, Object> values = getStringObjectMap((PrometheusMetricsSystem) metricsSystem);
+      metricList.add(extractBeaconNodeData(values));
+      metricList.add(extractValidatorData(values));
     } else {
       LOG.error("Prometheus metric system not found.");
+      metricList.add(
+          new MinimalMetricData(
+              PROTOCOL_VERSION, System.currentTimeMillis(), "minimal", metricsSystem));
     }
-    return new MinimalMetricData(
-        PROTOCOL_VERSION, System.currentTimeMillis(), "minimal", metricsSystem);
+    return metricList;
   }
 
-  private static BaseMetricData extractValidatorData(
-      PrometheusMetricsSystem prometheusMetricsSystem) {
+  private static BaseMetricData extractBeaconNodeData(Map<String, Object> values) {
+    final long disk_beaconchain_bytes_total;
+    final long network_libp2p_bytes_total_receive;
+    final long network_libp2p_bytes_total_transmit;
+    final int network_peers_connected;
+    final long sync_beacon_head_slot;
+    final String clientVersion;
+
+    if (values.containsKey("head_slot")) {
+      sync_beacon_head_slot =
+          ((Double) ((Observation) values.get("head_slot")).getValue()).longValue();
+    } else {
+      sync_beacon_head_slot = 0;
+    }
+    if (values.containsKey("teku_version")) {
+      clientVersion = ((Observation) values.get("teku_version")).getValue().toString();
+    } else {
+      clientVersion = "";
+    }
```

Comment:
Pretty sure `teku_version` will never be available from metrics.  You can just get it directly from `VersionProvider.VERSION` though.

Reply:
Great, I'll use the VersionProvider.IMPLEMENTATION_VERSION instead, because the VERSION property includes also the client name, version, OS and JVM.